Anime-App/Feature/Anime/NetworkingManager/NetworkingManager.swift

```diff
@@ -0,0 +1,78 @@
+//
+//  NetworkingManager.swift
+//  Anime-App
+//
+//  Created by Tobi Adegoroye on 12/08/2023.
+//
+
+import Foundation
+
+
+   enum NetworkingError: Error {
+       case invalidUrl
+       case custom(error: Error)
+       case invalidStatusCode(statusCode: Int)
+       case invalidData
+       case failedToDecode(error: Error)
+       case failedToReloadData
+   }
+
+
+class NetworkingManager {
+    static let shared = NetworkingManager()
+    var hasError = false
+    private(set) var isRefreshing =  false
+    private init() {}
+    
+    func request<T: Codable>(_ absoluteUrl: String,
+                             type: T.Type,
+                             completion: @escaping (Result<T,Error>) -> Void) {
+        isRefreshing = true
+
+        guard let url = URL(string: absoluteUrl) else {
+            completion(.failure(NetworkingError.invalidUrl))
+            return
+        }
+        
+        let request = URLRequest(url: url)
+        
+        let dataTask = URLSession.shared.dataTask(with: request) { data, response, error in
+            DispatchQueue.main.async {
+                
+                if error != nil {
+                    completion(.failure(NetworkingError.custom(error: error!)))
+                    return
+                }
+                
+                guard let response = response as? HTTPURLResponse,
+                      (200...300) ~= response.statusCode else {
+                    let statusCode = (response as! HTTPURLResponse).statusCode
+                    completion(.failure(NetworkingError.invalidStatusCode(statusCode: statusCode)))
+                    return
+                    
+                }
+                
+                
+                guard let data = data else {
+                    completion(.failure(NetworkingError.invalidData))
+                    return
+                }
+                
+                do {
+                    let decoder = JSONDecoder()
+                    decoder.keyDecodingStrategy = .convertFromSnakeCase
+                    let response = try decoder.decode(T.self, from: data)
+                    completion(.success(response))
+                } catch  {
+                    self.hasError = true
+                    completion(.failure(NetworkingError.failedToDecode(error: error)))
+                }
+                self.isRefreshing = false
```

Comment:
So you have a problem here, if the return statements get hit in the guard you will never set isRefreshing to false cos it won't reach this line.

What you should do is on line 41 you should use the defer keyword. Basically this makes sure that this is the last thing that is executed you use it like this 

defer { self.isLoading = false }

Also, i would rename the property to isLoading, it's not refreshing data which would mean that someone has pulled the scrollview down to refresh the page. What you're doing is loading data instead.